### PR TITLE
scan GitHub PR and issue comments

### PR DESCRIPTION
**Pull request by zubairk14**
Scan PR and issue comments. Support paginated calls to GitHub Issues and Pull Requests service. 

* https://docs.github.com/en/rest/pulls/comments?apiVersion=2022-11-28
* https://docs.github.com/en/rest/issues/comments?apiVersion=2022-11-28

### CODE REVIEW

File: pkg/sources/github/github.go [hunk 1]
```diff
@@ -919,3 +925,141 @@ func (s *Source) setProgressCompleteWithRepo(index int, offset int, repoURL stri
...
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case chunksChan <- chunk:
+		}
+	}
+	return nil
+}
+
+func removeURLAndSplit(url string) []string {
```
Comment (ahrav) [inline pkg/sources/github/github.go:1059]: suggestion: We might also want to ensure repos that are provided using ssh also works. I think we have a similar function that you can likely use as a reference: https://github.com/trufflesecurity/trufflehog/blob/452734adc8f92b723e33ac052c088544de131dc0/pkg/sources/git/git.go#L249

File: pkg/sources/github/github.go [hunk 2]
```diff
@@ -919,3 +925,141 @@ func (s *Source) setProgressCompleteWithRepo(index int, offset int, repoURL stri
...
+		Sort:      sortType,
+		Direction: directionType,
+		ListOptions: github.ListOptions{
+			PerPage: defaultPagination,
+			Page:    1,
+		},
+	}
+
+	for {
+		prComments, _, err := s.apiClient.PullRequests.ListComments(ctx, owner, repo, 0, prOpts)
```
Comment (ahrav) [inline pkg/sources/github/github.go:976]: suggestion: We should probably handle rate limits similar to how we do throughout this file.
ex: https://github.com/trufflesecurity/trufflehog/blob/452734adc8f92b723e33ac052c088544de131dc0/pkg/sources/github/github.go#L286

File: pkg/sources/github/github.go [hunk 3]
```diff
@@ -919,3 +925,141 @@ func (s *Source) setProgressCompleteWithRepo(index int, offset int, repoURL stri
...
 }
+
+func (s *Source) scanComments(ctx context.Context, repoPath string, chunksChan chan *sources.Chunk) error {
+	s.log.Info("scanning comments")
+
+	trimmedURL := removeURLAndSplit(repoPath)
+	owner := trimmedURL[1]
+	repo := trimmedURL[2]
+
+	sortType := "created"
```
Comment (ahrav) [inline pkg/sources/github/github.go:936]: nit: Grouping related vars tends to read a bit nicer imo.
ex:
```
	var (
		sortType      = "created"
		directionType = "desc"
	)
```

File: pkg/sources/github/github.go (hunk 2 above)
Comment (ahrav) [inline pkg/sources/github/github.go:976]: nit: Can we replace the magic '0' w/ a const and a tiny comment.
ex: `	const defaultIssueNumber = 0 // List all comments for all issues for the repo.`

File: pkg/sources/github/github.go [hunk 4]
```diff
@@ -919,3 +925,141 @@ func (s *Source) setProgressCompleteWithRepo(index int, offset int, repoURL stri
...
+		Sort:      &sortType,
+		Direction: &directionType,
+		ListOptions: github.ListOptions{
+			PerPage: defaultPagination,
+			Page:    1,
+		},
+	}
+
+	for {
+		issueComments, _, err := s.apiClient.Issues.ListComments(ctx, owner, repo, 0, issueOpts)
```
Comment (ahrav) [inline pkg/sources/github/github.go:949]: Whoops missed this one, but same comments as below.

**Review by ahrav — APPROVED** [review]
Nice job, LGTM!

File: pkg/sources/github/github.go (hunk 3 above)
Comment (zubairk14) [reply pkg/sources/github/github.go:936]: ahhh true

File: pkg/sources/github/github.go (hunk 2 above)
Comment (zubairk14) [reply pkg/sources/github/github.go:976]: yep for sure

File: pkg/sources/github/github.go (hunk 1 above)
Comment (zubairk14) [reply pkg/sources/github/github.go:1059]: oooh ill try this - thanks for the code pointer

File: pkg/sources/github/github.go (hunk 2 above)
Comment (zubairk14) [reply pkg/sources/github/github.go:976]: good call

File: pkg/sources/github/github.go (hunk 1 above)
Comment (zubairk14) [reply pkg/sources/github/github.go:1078]: just imported bill's gitURLParse (made it a public func)

**Pull request merged**